build_scripts/build_collect_diff

```diff
@@ -0,0 +1,158 @@
+#!/bin/bash
+#
+# Script to build the kernel, collect compiled file lists using modified kernel scripts,
+# and generate a git diff report based on the collected lists.
+
+set -e
+
+# check and install gcc-11 if not already installed
+install_package_safe() {
+    if ! command -v libssl-dev &> /dev/null; then
+        sudo apt-get update
+        sudo apt-get install -y libssl-dev
+    else
+        echo "libssl-dev is already installed."
+    fi
+}
```

Comment:
This function does not actually do anything useful. I would recommend removing it.

In particular, you probably do not need `libssl-dev` to compile the kernel, and `libssl-dev` is not a valid command, so this will always run the `apt` commands, even when `libssl-dev` is properly installed.

Reply:
Okay  I have removed the function!

Reply:
I think it was ever since ~4.x libssl-dev is needed as a dependency because of building signing tools.  I'll make a ticket to fix this host dependency as part of containerizing the script execution so dependencies are stable.